src/Model/Photo.php

```diff
@@ -1017,16 +1038,129 @@ public static function upload(int $uid, array $files)
 
 		$picture = [];
 
-		$picture['id']        = $photo['id'];
-		$picture['size']      = $photo['datasize'];
-		$picture['width']     = $photo['width'];
-		$picture['height']    = $photo['height'];
-		$picture['type']      = $photo['type'];
-		$picture['albumpage'] = DI::baseUrl() . '/photos/' . $user['nickname'] . '/image/' . $resource_id;
-		$picture['picture']   = DI::baseUrl() . '/photo/{$resource_id}-0.' . $Image->getExt();
-		$picture['preview']   = DI::baseUrl() . '/photo/{$resource_id}-{$smallest}.' . $Image->getExt();
+		$picture['id']          = $photo['id'];
+		$picture['resource_id'] = $resource_id;
+		$picture['size']        = $photo['datasize'];
+		$picture['width']       = $photo['width'];
+		$picture['height']      = $photo['height'];
+		$picture['type']        = $photo['type'];
+		$picture['albumpage']   = DI::baseUrl() . '/photos/' . $user['nickname'] . '/image/' . $resource_id;
+		$picture['picture']     = DI::baseUrl() . '/photo/{$resource_id}-0.' . $Image->getExt();
+		$picture['preview']     = DI::baseUrl() . '/photo/{$resource_id}-{$smallest}.' . $Image->getExt();
+
+		$Image->__destruct();
```

Comment:
Isn't this done by default when the function ends since it was created in the function?

Reply:
I just harmonized the different upload functionalities. When you say that it isn't needed then I will remove it.